uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Menu/MainMenuConfig.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+class CRM_HRCore_Menu_MainMenuConfig implements CRM_HRCore_Menu_Config {
```

Comment:
I have to see I didn't see the idea of every menu being a class coming. I have mentioned a configuration file multiple times, so this is what I was expecting. When I saw the `CRM_HRCore_Menu_Config` interface, I thought it would be implemented by some class that would read this configuration file rather than the holding the configuration itself. Could you give me more details about why you decided to implement it this way?

I do see the advantage of having a class representing SOME of the menus, like the Syleguide one, which is dynamic and has some logic in involved, but for most of the cases, a basic php file containing an array should have been enough

Reply:
tbh, the classes are not necessary for most of the menu config.
Agree basic php file will do for most of the menus and we don't really need the classes for most of them. Will make the non dynamic ones to be a php file returning an array rather than being classes.